samples/azure/performance-tuning-asb/ASB_7/FastAtomicReceiver/Program.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Threading.Tasks;
+using NServiceBus;
+using NServiceBus.Logging;
+
+class Program
+{
+    private static ILog _logger = LogManager.GetLogger<Program>();
+    public static ReceiveCounter ReceiveCounter = new ReceiveCounter();
+
+    static void Main()
+    {
+        MainAsync().GetAwaiter().GetResult();
+    }
+
+    static async Task MainAsync()
+    {
+        Console.Title = "Samples.ASB.Performance.FastAtomicReceiver";
+
+        ReceiveCounter.Subscribe(i => _logger.Warn("Process " + i + " messages per second"));
+
+        var endpointConfiguration = new EndpointConfiguration("Samples.ASB.Performance.Receiver");
+        var transportConfiguration = endpointConfiguration.UseTransport<AzureServiceBusTransport>();
+        var connectionString = Environment.GetEnvironmentVariable("AzureServiceBus.ConnectionString");
+
+        if (string.IsNullOrWhiteSpace(connectionString))
+        {
+            throw new Exception("Could not read the 'AzureServiceBus.ConnectionString' environment variable. Check the sample prerequisites.");
+        }
+        transportConfiguration.ConnectionString(connectionString);
+        var topology = transportConfiguration.UseTopology<ForwardingTopology>();
+
+        endpointConfiguration.SendFailedMessagesTo("error");
+        endpointConfiguration.UseSerialization<JsonSerializer>();
+        endpointConfiguration.EnableInstallers();
+        endpointConfiguration.UsePersistence<InMemoryPersistence>();
+
+        #region fast-atomic-receiver-config
```

Comment:
I'll put it in text, otherwise need to adjust all the visual studio project names, assembly names etc...